### ISSUE TITLE
Cannot update netty pipeline with outbound channel handlers

### ISSUE THREAD

**Issue by stijnopheide** [ztellman/aleph#543]
Currently transforming the netty pipeline with outbound channel handlers on a TCP server does not work. Aleph expects that you only put onto the stream something that can be converted to a ByteBuf by `aleph.netty/to-byte-buf`. 

For incoming channel handlers you can change this behavior by setting `:raw-stream? true`, however that does not change the behavior on outbound channel handlers. See https://github.com/ztellman/aleph/blob/master/src/aleph/tcp.clj#L47.

Probably this same problem exists for TCP clients.

The use case is adding frame splitters/creators for Protobuf messages (https://netty.io/4.1/api/io/netty/handler/codec/protobuf/ProtobufDecoder.html) which is implemented as netty encoders/decoders. In the TCP handler you would receive a decoded Protobuf message and you have to return a Protobuf message, not a coercable-to-ByteBuf.

**Comment by KingMob** [comment]
Yeah, by and large, Aleph _uses_ Netty, but doesn't support _you_ using Netty very well. There may have been good reasons for this (like the ref-counting management required when using your own channel handlers), but I'm not privy to Zach's mind.

The general expectation is the main handler you supply takes care of everything. It would be nice to better support channel handlers, though, so if you want to make a PR for UDP and TCP that adds a `:coerce-fn` key to the options maps, I'd support that.

**Comment by arnaudgeiser** [comment]
Starting to work on this.
We worked a lot with GRPC (and thus Protobuf) during last year, I consider this feature request valuable.

**Comment by ohbadiah** [comment]
@arnaudgeiser is there a branch I could take a look at?

**Comment by ohbadiah** [comment]
I mentioned this issue in Clojurians slack [here](https://clojurians.slack.com/archives/C0G922PCH/p1702068332699309).

**Comment by arnaudgeiser** [comment]
I don't have much but the following branch : https://github.com/clj-commons/aleph/compare/protobuf?expand=1
